File: design-patterns/go/prototype/transport/plane.go
// plane.go

package main

import "fmt"

type Plane struct {
	Model string
	Color string
}

func NewPlane() (plane *Plane) {
	plane = &Plane{}
	return
}

func (plane *Plane) Clone() Prototype {
	planeClone := NewPlane()
	planeClone.Model = plane.Model
	planeClone.Color = plane.Color

	return planeClone
}

func (plane *Plane) ToString() string {
	return fmt.Sprintf("Model: %v | Color: %v", plane.Model, plane.Color)
}

File: design-patterns/ts/prototype/transport/plane.ts
// plane.ts

import Prototype from "./prototype";


class Plane implements Prototype {
    model: string;
    color: string;

    constructor(model: string, color: string) {
        this.model = model;
        this.color = color;
    }

    clone(): Prototype {
        return Object.assign({}, this)
    }

    toString(): string {
        return "Model: " + this.model + " | Color: " + this.color;
    }
}

export default Plane;